### PR TITLE
Bugfixes for python-nextgen

### PR DESCRIPTION
**Pull request by tom300z**
- fixed swapped operators for conint typing
- added string conversion to support non-string url params 

@wing328 Mention since you seem to be the main python-nextgen dev. Nice work btw.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
Tested locally updated samples but got errors:
```
_____________________ PetApiTests.test_find_pets_by_status _____________________

self = <tests.test_pet_api.PetApiTests testMethod=test_find_pets_by_status>

    def test_find_pets_by_status(self):
        self.pet_api.add_pet(self.pet)
        self.pet_api.add_pet(self.pet2)
    
        results = self.pet_api.find_pets_by_status(status=["pending", "available"])
        self.assertIsInstance(results, list)
>       self.assertTrue(len(results) > 1)
E       AssertionError: False is not true

tests/test_pet_api.py:217: AssertionError
______________________ PetApiTests.test_find_pets_by_tags ______________________

self = <tests.test_pet_api.PetApiTests testMethod=test_find_pets_by_tags>

    def test_find_pets_by_tags(self):
        self.pet_api.add_pet(self.pet)
    
        self.assertIn(
            self.pet.id,
>           list(map(lambda x: getattr(x, 'id'), self.pet_api.find_pets_by_tags(tags=[self.tag.name, "here is another tag"])))
        )
E       AssertionError: 1518215766 not found in []
```

can you please take a look when you've time?

https://github.com/OpenAPITools/openapi-generator/wiki/Integration-Tests contains instructions on how to run the petstore server locally.

File: modules/openapi-generator/src/main/resources/python-nextgen/api_client.mustache [hunk 1]
```diff
@@ -563,7 +563,7 @@ class ApiClient(object):
             else:
                 new_params.append((k, v))
 
-        return "&".join(["=".join(item) for item in new_params])
+        return "&".join(["=".join(str(item)) for item in new_params])
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/python-nextgen/api_client.mustache:566]: can you please elaborate on the fix for this line? can you share a spec to more easily reproduce the issue fixed by this change?

**Comment by tom300z** [comment]
Thanks for the tips first of all. I've updated the signature of my commits to use my github email, hopefully that should link them to my account. 
I'll take a look at the failing petstore test in a moment.

**Comment by tom300z** [comment]
I've updated the samples and provided a better fix that doesn't break any tests.

The fix in [Provide better fix for nonstring url param values](https://github.com/OpenAPITools/openapi-generator/pull/14334/commits/038623ed84e298c1b16ce2e033584364d3098fa9) is needed when there are non-string query params in an endpoint. 
here's an example schema: 
```json
{
  "swagger": "2.0",
  "info": {
    "description": "VMware Cloud Director OpenAPI is a new API that is defined using the OpenAPI standards.<br/>\nThis ReSTful API borrows some elements of the legacy VMware Cloud Director API and establishes new patterns for use\nas described below.\n<h4>Authentication</h4>\nAuthentication and Authorization schemes are the same as those for the legacy APIs. You can authenticate\nusing the JWT token via the <code>Authorization</code> header or specifying a session using\n<code>x-vcloud-authorization</code> (The latter form is deprecated).\n<h4>Operation Patterns</h4>\nThis API follows the following general guidelines to establish a consistent CRUD pattern:\n<table>\n<tr>\n  <th>Operation</th><th>Description</th><th>Response Code</th><th>Response Content</th>\n</tr><tr>\n  <td>GET /items<td>Returns a paginated list of items<td>200<td>Response will include Navigational links to the items in the list.\n</tr><tr>\n  <td>POST /items<td>Returns newly created item<td>201<td>Content-Location header links to the newly created item\n</tr><tr>\n  <td>GET /items/urn<td>Returns an individual item<td>200<td>A single item using same data type as that included in list above\n</tr><tr>\n  <td>PUT /items/urn<td>Updates an individual item<td>200<td>Updated view of the item is returned\n</tr><tr>\n  <td>DELETE /items/urn<td>Deletes the item<td>204<td>No content is returned.\n</tr>\n</table>\n<h5>Asynchronous operations</h5>\nAsynchronous operations are determined by the server. In those cases, instead of responding as described above,\nthe server responds with an HTTP Response code 202 and an empty body. The tracking task (which is the same task as\nall legacy API operations use) is linked via the URI provided in the <code>Location</code> header.<br/>\nAll API calls can choose to service a request asynchronously or synchronously as determined by the server upon\ninterpreting the request. Operations that choose to exhibit this dual behavior will have both options documented by\nspecifying both response code(s) below. The caller must be prepared to handle responses to such API calls by\ninspecting the HTTP Response code.\n<h5>Error Conditions</h5>\n<b>All</b> operations report errors using the following error reporting rules:\n<ul>\n  <li>400: Bad Request - In event of bad request due to incorrect data or other user error</li>\n  <li>401: Bad Request - If user is unauthenticated or their session has expired</li>\n  <li>403: Forbidden - If the user is not authorized or the entity does not exist</li>\n</ul>\n<h4>OpenAPI Design Concepts and Principles</h4>\n<ul>\n  <li>IDs are full Uniform Resource Names (URNs).</li>\n  <li>OpenAPI's <code>Content-Type</code> is always <code>application/json</code></li>\n  <li>REST links are in the Link header.</li>\n  <ul>\n    <li>Multiple relationships for any link are represented by multiple values in a space-separated list.</li>\n    <li>Links have a custom VMware Cloud Director-specific &quot;model&quot; attribute that hints at the applicable data\n        type for the links.</li>\n    <li>title + rel + model attributes evaluates to a unique link.</li>\n    <li>Links follow Hypermedia as the Engine of Application State (HATEOAS) principles. Links are present if\n        certain operations are present and permitted for the user&quot;s current role and the state of the\n        referred entities.</li>\n  </ul>\n  <li>APIs follow a flat structure relying on cross-referencing other entities instead of the navigational style\n      used by the legacy VMware Cloud Director APIs.</li>\n  <li>Most endpoints that return a list support filtering and sorting similar to the query service in the legacy\n      VMware Cloud Director APIs.</li>\n  <li>Accept header must be included to specify the API version for the request similar to calls to existing legacy\n      VMware Cloud Director APIs.</li>\n  <li>Each feature has a version in the path element present in its URL.<br/>\n      <b>Note</b> API URL's without a version in their paths must be considered experimental.</li>\n</ul>\n",
    "version": "36.3",
    "title": "VMware Cloud Director OpenAPI"
  },
  "basePath": "/cloudapi",
  "paths": {
    "/1.0.0/edgeGateways/{gatewayId}/usedIpAddresses": {
      "get": {
        "tags": [
          "edgeGateway"
        ],
        "summary": "Retrieve the list of IP addresses which are being used by the edge gateway.",
        "description": "Get all the IP Addresses which are being used by the Edge Gateway such as the primary IP or an IP used by a given\nEdge Service, such as NAT. These IP addresses are a subset of the IPs allocated from the connected external networks.\nIf the IP is  being consumed by any of the configured services on the edge gateway then name of service will be returned.\n",
        "operationId": "getUsedIpAddresses",
        "produces": [
          "application/json;version=36.3"
        ],
        "parameters": [
          {
            "name": "filter",
            "in": "query",
            "description": "Filter for a query.  FIQL format.",
            "required": false,
            "type": "string"
          },
          {
            "name": "sortAsc",
            "in": "query",
            "description": "Field to use for ascending sort",
            "required": false,
            "type": "string"
          },
          {
            "name": "sortDesc",
            "in": "query",
            "description": "Field to use for descending sort",
            "required": false,
            "type": "string"
          },
          {
            "name": "page",
            "in": "query",
            "description": "Page to fetch, zero offset.",
            "required": false,
            "type": "integer",
            "default": 1,
            "minimum": 1
          },
          {
            "name": "pageSize",
            "in": "query",
            "description": "Results per page to fetch.",
            "required": false,
            "type": "integer",
            "default": 25,
            "maximum": 128,
            "minimum": 0
          },
          {
            "name": "gatewayId",
            "in": "path",
            "required": true,
            "type": "string",
            "x-vcloud-constraints": [
              {
                "constraint": "Urn",
                "args": [
                  {
                    "arg": "gateway"
                  }
                ]
              }
            ]
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    }
  }
}
```
integer parameters like "page" or "pageSize" would cause an error without the fix. Let me know if you need more details

File: modules/openapi-generator/src/main/resources/python-nextgen/api_client.mustache [hunk 2]
```diff
@@ -545,6 +545,11 @@ class ApiClient(object):
         if collection_formats is None:
             collection_formats = {}
         for k, v in params.items() if isinstance(params, dict) else params:  # noqa: E501
+            if isinstance(v, (int, float)):
+                v = str(v)
+            if isinstance(v, bool):
+                v = str(v).lower()
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/python-nextgen/api_client.mustache:552]: I'll remove spaces in this line in another PR.

**Comment by wing328** [comment]
Local tests with updated samples passed without issues. Thanks for the PR.

File: modules/openapi-generator/src/main/resources/python-nextgen/api_client.mustache (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/python-nextgen/api_client.mustache:552]: Filed https://github.com/OpenAPITools/openapi-generator/pull/14347

**Pull request merged**